Foundry V10 compatibility

joevaughan opened the pull request.

joevaughan commented:
As discussed in issue #39 - A PR for Foundry V10 compatibility (and a few tweaks for some other things like making sure the Compendium Folders modules' hidden actors don't show in the NPC Browser)

spetzel2020 commented on compendium-browser.js:
```diff
@@ -980,12 +1015,23 @@ class CompendiumBrowser extends Application {
         item.damageTypes = [];
-        if (item.data.damage && item.data.damage.parts.length > 0) {
-            for (let part of item.data.damage.parts) {
+        if (item.system.damage && item.system.damage.parts.length > 0) {
```
I might be reading this wrong, but is item.system.damage here within a test for v10? If not, then won't this fail for v0.8? I thought that the switch to item.system was introduced in v10, but I'm quite possibly wrong.

See my top-level comment about setting minimum-compatible version to be v10

spetzel2020 commented on module.json:
```diff
@@ -57,7 +61,7 @@
     "manifest": "https://github.com/League-of-Foundry-Developers/compendium-browser/releases/latest/download/module.json",
     "download": "https://github.com/League-of-Foundry-Developers/compendium-browser/releases/download/v0.7.2/compendium-browser.zip",
     "minimumCoreVersion": "0.8.6",
```
Can I suggest that until this version is confirmed to work in v10, you change the `minimumCoreVersion` to 10? Just in case your v10 changes break the (currently working version) for Foundry 0.8.

spetzel2020 reviewed (changes requested):
Made a couple of comments based on a code review but not on actual testing (with either Foundry v0.8 or v10).

You're currently much closer to this than I am, so feel free to ignore; just reply accordingly and I will merge the PR.

joevaughan commented:
> 

Good catch. I didn't consider every possible area of change between 8, 9 and 10. 

I've reworked that section (and some other sections) and reworked the filters which I discovered didn't work properly. 

However, I've set this to v9+ compatibility (because earliest version I could properly test with was 9.266 - Cant even get v8 Game systems etc. without jumping through hoops (can't install via the Foundry UI) as I have been unable to test v0.8.6

Have tested that this works on v9.266+ (and v10 of course)

spetzel2020 commented:
Thanks for this! As soon as I have a chance later I will release the new 0.8 version (I'm away, and in a different timezone, so sorry for the delays)

(And I completely agree with your decision/comments about supporting Foundry v0.8)
